deepchem/models/torch_models/torch_model.py

```diff
@@ -160,6 +161,8 @@ def __init__(self,
     device: torch.device
       the device on which to run computations.  If None, a device is
       chosen automatically.
+    collate_fn: Function, default None
+      This function makes specific batch data for each models.
```

Comment:
A few issues with this:

1. We should try to keep the TorchModel and KerasModel APIs identical as much as possible, so if we add a new argument to one we should also add it to the other.
2. The name `collate_fn` doesn't mean anything to me, and doesn't seem to have anything to do with what it does.  I suspect the name is specific to how you're using it in a particular model?
3. The description isn't useful either.  Saying that it "makes specific batch data" leaves the reader confused about what it does.
4. The normal way we handle cases like this is by overriding `default_generator()`.

Reply:
> 2. The name collate_fn doesn't mean anything to me, and doesn't seem to have anything to do with what it does. I suspect the name is specific to how you're using it in a particular model?
> 3. The description isn't useful either. Saying that it "makes specific batch data" leaves the reader confused about what it does.

I'm so sorry. The docstring is still WIP. I definitely update them. The `collate_fn`  is derived from `collate_fn` of DataLoader in PyTorch. The `collate_fn` argument is often used in DataLoader in order to make user-defined batch.

Please see https://pytorch.org/docs/stable/data.html#dataloader-collate-fn

> 4.The normal way we handle cases like this is by overriding default_generator()

The main problem is that the line of `torch.as_tensor(x, device=self.device)` in `self._prepare_batch` doesn't work with `GraphData` and DGLGraph. Can I fix this problem by overriding default_generator()?

Reply:
The `collate_fn` in DataLoader is used for something very different.  It's part of automatic batching, and it determined how a group of individual samples get combined into a batch.  By the time the model sees them, they've already been combined.  That's why it's a property of DataLoader, not torch.nn.Module.

> The main problem is that the line of torch.as_tensor(x, device=self.device) in self._prepare_batch doesn't work with GraphData and DGLGraph. Can I fix this problem by overriding default_generator()?

Take a look at `GraphConvModel.default_generator()`.  It does exactly this, converting a set of ConvMol objects from the dataset into arrays that can be passed to the model.

Reply:
> The collate_fn in DataLoader is used for something very different. It's part of automatic batching, and it determined how a group of individual samples get combined into a batch. By the time the model sees them, they've already been combined. That's why it's a property of DataLoader, not torch.nn.Module.

OK.

> Take a look at GraphConvModel.default_generator(). It does exactly this, converting a set of ConvMol objects from the dataset into arrays that can be passed to the model.

We cannot convert DGLGraph to arrays because the general DGL models (of course, contains CGCNNLayer) use DGLGraph's method for updating node or edge representations. I overrided `_prepare_batch` method.